### PR TITLE
CXX-685: Support Filtered (Partial) Indexes

### PR DESCRIPTION
**Pull request by markbenvenuto**
Add support for Filtered (Partial) Indexes

See https://github.com/mongodb/specifications/blob/master/source/index-management.rst for the SPEC

### CODE REVIEW

File: src/mongo/integration/standalone/dbclient_test.cpp [hunk 1]
```diff
@@ -1377,6 +1377,14 @@ TEST_F(DBClientTest, CreateHashedIndex) {
     c->createIndex(TEST_NS, IndexSpec().addKey("aField", IndexSpec::kIndexTypeHashed));
 }
 
+TEST_F(DBClientTest, CreatePartialIndex) {
+    if (serverGTE(c.get(), 3, 1)) {
+        c->createIndex(
+            TEST_NS,
+            IndexSpec().addKey("aField").partialFilterExpression(BSON("aField" << GT << 5)));
+    }
```
Comment (amidvidy) [inline src/mongo/integration/standalone/dbclient_test.cpp]: can you check with getIndexInfos that the server's index spec has the expected partialFilterExpression field?

**Comment by amidvidy** [comment]
LGTM

**Comment by acmorrow** [comment]
LGTM

**Comment by acmorrow** [comment]
@markbenvenuto Please remember to commit this on the new legacy-1.1-dev branch, not on the primary 'legacy' branch.

**Comment by markbenvenuto** [comment]
Committed with https://github.com/mongodb/mongo-cxx-driver/commit/342b0bb39612f86abc326076c893dc35f1fc630c

**Pull request closed without merging**